Source/WebCore/Modules/compression/CompressionStream.js

```diff
@@ -66,7 +73,13 @@ function initializeCompressionStream(format)
     };
     const flushAlgorithm = () => {
         const encoder = @getByIdDirectPrivate(this, "CompressionStreamEncoder");
-        const buffer = encoder.@flush();
+
+        let buffer;
+        try {
+            buffer = encoder.@flush();
+        } catch (e) {
+            return @Promise.@reject(@makeTypeError(e.message));
+        }
         if (buffer) {
             const transformStream = @getByIdDirectPrivate(this, "CompressionStreamTransform");
             const controller = @getByIdDirectPrivate(transformStream, "controller");
```

Comment:
It seems like it would be simpler just to have this entire block inside the 'try' section, rather than declaring 'buffer' outside, since it's never used again after these steps.